client/src/pages/SignUp.js

```diff
@@ -0,0 +1,238 @@
+        } else {
+          setPasswordError("");
+        }
+        setPassword(e.target.value);
+        break;
+      }
+      case "signupPwdConfirm": {
+        if (e.target.value !== password) {
+          setPasswordConfirmError("Passwords not matching");
+        } else {
+          setPasswordConfirmError("");
+        }
+        break;
+      }
+      default:
+        break;
+    }
+  };
+  return (
+    <Grid container direction="row" className={classes.signUpContainer}>
+      <Grid item md={6} sm={12}>
+        <Grid
+          container
+          direction="column"
+          justify="flex-start"
+          alignItems="center"
+        >
+          <Grid item className={classes.titleLogo}></Grid>
+          <Grid item>
+            <Card>
+              <CardContent>
+                <Typography className={classes.title} gutterBottom>
+                  Create an account
+                </Typography>
+                <form autoComplete="off">
+                  <Grid
+                    container
+                    direction="column"
+                    justify="flex-start"
+                    alignItems="center"
+                  >
+                    <Grid item>
+                      <TextField
+                        className={classes.textField}
+                        required
+                        autoFocus
+                        name="signupName"
+                        label="Name"
+                        value={username}
+                        placeholder="Enter your name"
+                        error={usernameError !== "" ? true : false}
+                        variant="outlined"
+                        onChange={(e) => validateChange(e)}
+                        helperText={usernameError}
+                      />
+                    </Grid>
+                    <Grid item>
+                      <TextField
+                        className={classes.textField}
+                        required
+                        name="signupEmail"
+                        label="Email"
+                        value={email}
+                        placeholder="Enter your e-mail address"
+                        variant="outlined"
+                        error={emailError !== "" ? true : false}
+                        helperText={emailError}
+                        onChange={(e) => validateChange(e)}
+                      />
+                    </Grid>
+                    <Grid item>
+                      <TextField
+                        className={classes.textField}
+                        required
+                        type="password"
+                        name="signupPwd"
+                        label="Password"
+                        value={password}
+                        error={passwordError !== "" ? true : false}
+                        placeholder="Enter password"
+                        variant="outlined"
+                        helperText="Minimum 6 characters"
```

Comment:
This helper text reads like an error to the user, I think it's better to just use the error message to enforce length